pkg/utils/k8sutil_test.go

```diff
@@ -73,3 +75,51 @@ func TestDeleteK8sResources(t *testing.T) {
 		t.Errorf("failed to delete service")
 	}
 }
+
+func TestEnsureK8sResources(t *testing.T) {
+	clientset := fake.NewSimpleClientset()
+	ns := "myns"
+	funcName := "foo"
+
+	funcLabels := map[string]string{
+		"foo": "bar",
+	}
+
+	f := &spec.Function{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Function",
+			APIVersion: "k8s.io/v1",
+		},
+		Metadata: metav1.ObjectMeta{
+			Name:      funcName,
+			Namespace: ns,
+			Labels:    funcLabels,
+		},
+		Spec: spec.FunctionSpec{
+			Handler:  "foo.bar",
+			Runtime:  "",
+			Type:     "",
+			Function: "",
+			Topic:    "HTTP",
+			Deps:     "",
+			Desc:     "",
+			Template: v1.PodTemplateSpec{
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Env: []v1.EnvVar{},
+							Resources: v1.ResourceRequirements{
+								Limits:   map[v1.ResourceName]resource.Quantity{},
+								Requests: map[v1.ResourceName]resource.Quantity{},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	if err := EnsureK8sResources(ns, funcName, f, clientset); err != nil {
+		t.Fatalf("Creating resources returned err: %v", err)
+	}
+}
```

Comment:
You should also check that the clientset has recorded the actions you expected to see - or else just look in the fake's ObjectTracker for the expected server-side objects.